pytta/rooms.py

```diff
@@ -28,13 +33,267 @@
 
 import numpy as np
 import matplotlib.pyplot as plt
-from numba import njit
+# from numba import njit
 from pytta import SignalObj, OctFilter, Analysis, ImpulsiveResponse
-from pytta.utils import fractional_octave_frequencies as FOF
+from pytta.utils import fractional_octave_frequencies as FOF, freq_to_band
 import traceback
 import copy as cp
 
 
+class RoomParameters(Analysis):
+    """Room monaural parameters."""
+
+    def __init__(self, ir: SignalObj, nthOct: int = 1,
+                 minFreq: float = 2e1, maxFreq: float = 2e4, *args,
+                 bypassLundeby: bool = False, suppressWarnings: bool = True,
+                 ircut: float = None, **kwargs):
+        """
+        Room acoustical parameters for quality analysis.
+
+        Provides interface to estimate several room parameters based on
+        the energy distribution of the impulse response.
+
+        Parameters
+        ----------
+        ir : SignalObj
+            Monaural room impulse response signal.
+        nthOct : int, optional
+            Number of bands per octave. The default is 1.
+        minFreq : float, optional
+            Central frequency of the first band. The default is 2e1.
+        maxFreq : float, optional
+            Central frequency of the last band. The default is 2e4.
+        *args : Tuple
+            See Analysis.
+        bypassLundeby : bool, optional
+            Bypass Lundeby calculation, or not. The default is False.
+        suppressWarnings : bool, optional
+            Supress Lundeby warnings. The default is True.
+        ircut : float, optional
+            Cut the IR and throw away the silence tail. The default is None.
+        **kwargs : Dict
+            See Analysis.
+
+        Returns
+        -------
+        None.
+
+        """
+        _ir = ir.IR if type(ir) == ImpulsiveResponse else ir
+        minBand = freq_to_band(minFreq, nthOct, 1000, 10)
+        maxBand = freq_to_band(maxFreq, nthOct, 1000, 10)
+        nbands = maxBand - minBand + 1
+        super().__init__('mixed', nthOct, minFreq, maxFreq, nbands*[0], *args, **kwargs)
+        self.ir = crop_IR(_ir, ircut)
```

Comment:
Here you crop the IR for the first time, and looks the algorithm used in this function is quite similar to the one used in the Lundeby algorithm, so for the RT parameter, the IR is being overcut. It could be error-prone.

Reply:
These are the same steps made by the `analyse` function, I might look at it for a complete rewriting, but would take a bit more of time.